extension.js

```diff
@@ -45,45 +44,78 @@ scalar BigInt
 scalar Date
 scalar Any
 `;
-let resolvers = {};
-let apollo_options;
-let apollo;
-export function start(options) {
-	apollo_options = options;
-}
-export async function handleFile(content, url_path, file_path, resources) {
-	if (file_path.endsWith('.graphql'))
-		graphql_schema += content;
-	if (file_path.endsWith('.js')) {
-		const module_url = pathToFileURL(file_path).toString();
-		// load JS file and assign to resolvers
-		const module_exports = await import(module_url);
-		Object.assign(resolvers, module_exports.default || module_exports);
+
+export function start(options = {}) {
+	const config = {
+		cache: options.cache,
+		port: options.port,
+		resolvers: options.resolvers ?? './resolvers.js',
+		schemas: options.schemas ?? './schemas.graphql',
+		securePort: options.securePort,
 	}
-}
-export async function ready() {
-	if (isMainThread || apollo) return;
-	apollo = new ApolloServer({
-		typeDefs: graphql_schema,
-		resolvers,
-		cache: new HarperDBCache()
-	})
-	await apollo.start();
-	server.http(async (request, next_handler) => {
-		if (request.url === '/graphql') {
-			let body = await streamToBuffer(request.body);
-			request = Object.assign({}, request);
-			request.body = JSON.parse(body);
-			request.search = parse(request.url).search || '';
-			let response = await apollo.executeHTTPGraphQLRequest({
-				httpGraphQLRequest: request,
-				context: () => request,
+
+	logger.debug('@harperdb/apollo extension configuration:\n' + JSON.stringify(config, null, 2));
+
+	return {
+		async handleDirectory(_, componentPath) {
+
+			// Load the resolvers
+			const resolversPath = join(componentPath, config.resolvers);
+			const resolvers = await import(pathToFileURL(resolversPath));
+
+			// Load the schemas
+			// posix.join is necessary so that `/` are retained, otherwise they get normalized to the platform
+			const schemasPath = posix.join(componentPath, config.schemas)
```

Comment:
NIT but nbd obvs: missing ;

Reply:
Good catch. I should probably add a formatter. Ill do that as a follow up.